### PR TITLE
Add script for deploying/testing a cluster on GCE

### PR DESCRIPTION
**Pull request by benmoss**
Depends on windows-tools and k/k, assumes gcloud has already been configured. Probably needs another wrapper script to deal with Boskos

@neolit123

### CODE REVIEW

File: kubeadm/hack/e2e-cluster-gcp.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+kubectl -n kube-system wait --for=condition=Ready pods --all --timeout=15m
+
+# setting this env prevents ginkgo e2e from trying to run provider setup
+export KUBERNETES_CONFORMANCE_TEST="y"
+
+$(go env GOPATH)/src/sigs.k8s.io/windows-testing/gce/run-e2e.sh \
+	--provider=skeleton --num-nodes=$(expr $NODE_COUNT + 1) \
+	--ginkgo.focus="$FOCUS" --ginkgo.skip="$SKIP" \
+	--node-os-distro=windows --report-dir="$ARTIFACTS" \
+	--disable-log-dump=true
```
Comment (neolit123) [inline kubeadm/hack/e2e-cluster-gcp.sh]:
```
  # ginkgo regexes
  SKIP="${SKIP:-}"
  FOCUS="${FOCUS:-"\\[Conformance\\]"}"
  # if we set PARALLEL=true, skip serial tests set --ginkgo-parallel
  if [ "${PARALLEL:-false}" = "true" ]; then
    export GINKGO_PARALLEL=y
    if [ -z "${SKIP}" ]; then
      SKIP="\\[Serial\\]"
    else
      SKIP="\\[Serial\\]|${SKIP}"
    fi
  fi
```

Comment (neolit123) [reply kubeadm/hack/e2e-cluster-gcp.sh]: $NODE_COUNT should be N workers

Comment (neolit123) [reply kubeadm/hack/e2e-cluster-gcp.sh]: as mentioned on slack i think we should have something like that here above the `run-e2e.sh` call:
```
# split k8sVersion into major, minor
pushd $(go env GOPATH)/src/k8s.io/kubernetes/kubernetes
git fetch https://github.com/kubernetes/kubernetes.git release-$major.$minor
git checkout release-$major.$minor
popd
```

logs of what prow does:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-1-17/1237332976740601859/clone-log.txt

File: kubeadm/hack/startup/controlplane.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+apt-get update && apt-get install -y apt-transport-https curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+cat <<EOF | tee /etc/apt/sources.list.d/kubernetes.list
+deb https://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+apt-get update
+apt-get install -y kubelet kubeadm kubectl
+apt-mark hold kubelet kubeadm kubectl
+
+externalIP=$(curl -H "Metadata-Flavor:Google" 169.254.169.254/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip)
```
Comment (neolit123) [inline kubeadm/hack/startup/controlplane.sh]: wondering what this line does and where is `169.254.169.254` coming from?
would this hardcoded IP work for the Boskos setup?

Comment (benmoss) [reply kubeadm/hack/startup/controlplane.sh]: This is the GCP metadata server, same as `metadata.google.internal` maybe that's clearer. Should work from any GCP VM.

Comment (neolit123) [reply kubeadm/hack/startup/controlplane.sh]: ok, got it.

**Pull request merged**